### PR TITLE
SNOW-1367208: fix No stored procedure will run in a task unless pin snowflake-snowpark-python to an earlier version.

### PR DESCRIPTION
**Pull request by sfc-gh-yuwang**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1367208

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/_internal/server_connection.py [hunk 1]
```diff
@@ -541,6 +544,7 @@ def get_result_set(
         data_type: _AsyncResultType = _AsyncResultType.ROW,
         log_on_exception: bool = False,
         case_sensitive: bool = True,
+        ignore_results: bool = True,
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/_internal/server_connection.py:547]: nit: this should be `False` by default

Comment (sfc-gh-yuwang) [reply src/snowflake/snowpark/_internal/server_connection.py]: updated

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@
 
 #### Improvements
 
+#### Bug Fixes
+
+- Fixed a bug that no stored procedure will run in a task because `DataFrame._execute_and_get_query_id` tries to get query results.
```
Comment (sfc-gh-aalam) [inline CHANGELOG.md:11]:
```suggestion
- Fixed a bug where python stored procedure with table return type fails when run in a task.
```

**Review by sfc-gh-aalam — APPROVED** [review]
looks good with one doc related change

**Review by sfc-gh-jrose — APPROVED** [review]

**Pull request merged**